src/fixedSizeSquareMatrixOpsImpl.hpp

```diff
@@ -529,6 +529,56 @@ struct SquareMatrixOps< 2 >
     dstMatrix[ 1 ][ 0 ] = srcSymMatrix[ 2 ];
   }
 
+  /**
+   * @brief Determine the polar decomposition of the 2x2 matrix @p matrix
+   * @tparam DST_MATRIX The type of @p R.
+   * @tparam MATRIX The type of @p matrix.
+   * @param R The resultant rotation matrix.
+   * @param matrix The matrix to be decomposed.
+   * @details The polar decomposition returns a rotation matrix such that @p R . U = V . @p R = @p matrix.
+   *   This is done using Higham's iterative algorithm.
+   */
+  template< typename DST_MATRIX, typename MATRIX >
+  LVARRAY_HOST_DEVICE CONSTEXPR_WITHOUT_BOUNDS_CHECK inline
+  static void polarDecomposition( DST_MATRIX && LVARRAY_RESTRICT_REF R,
+                                  MATRIX const & LVARRAY_RESTRICT_REF matrix )
+  {
+    checkSizes< 2, 2 >( R );
+    checkSizes< 2, 2 >( matrix );
+
+    using FloatingPoint = std::decay_t< decltype( matrix[ 0 ][ 0 ] ) >;
+
+    // Initialize
+    copy< 2, 2 >( R, matrix );
+    FloatingPoint RInverse[2][2] = { {0} },
+                  RInverseTranspose[2][2] = { {0} },
+                  RRTMinusI[2][2] = { {0} };
+
+    // Higham Algorithm
+    FloatingPoint error = 1.0;
+    int iter = 0;
+    while( error > 1.0e-16 && iter < 100 )
+    {
+      iter++;
+      error = 0.0;
+
+      invert( RInverse, R );
+      transpose< 2, 2 >( RInverseTranspose, RInverse );
+      add< 2, 2 >( R, RInverseTranspose );
+      scale< 2, 2 >( R, 0.5 );
+      Rij_eq_AikBjk< 2, 2, 2 >( RRTMinusI, R, R );
+      addIdentity< 2 >( RRTMinusI, -1.0 );
+
+      for( std::ptrdiff_t i = 0 ; i < 2 ; i++ )
+      {
+        for( std::ptrdiff_t j = 0 ; j < 2 ; j++ )
+        {
+          error += RRTMinusI[i][j] * RRTMinusI[i][j];
+        }
+      }
+    }
```

Comment:
Do you want to error out (or at least return a signal) if it doesn't converge?

Reply:
What should I use to print an error message? Using LVARRAY_WARNING_IF() fails to compile since it calls std::cout.

Reply:
I'd either return the achieved tolerance or the number of iterations or just a boolean to indicate success or failure. Alternatively you can use `LVARRAY_ERROR_IF` which works on device to just kill the program.